### PR TITLE
add alibabacloud ai search notebook

### PR DESCRIPTION
**Pull request by Huaixinww**
Related to https://www.elastic.co/guide/en/elasticsearch/reference/master/infer-service-alibabacloud-ai-search.html
This PR adds a Jupyter notebook that contains an end-to-end example of using the Inference API with the AlibabaCloud AI Search service.

### CODE REVIEW

**Comment by serenachou** [comment]
@elastic/search-experiences-team  please review

File: notebooks/alibabacloud-ai-search/inference-alibabacloud-ai-search.ipynb
Comment (joemcelroy) [inline notebooks/alibabacloud-ai-search/inference-alibabacloud-ai-search.ipynb:1]: _Commented on notebook `notebooks/alibabacloud-ai-search/inference-alibabacloud-ai-search.ipynb` Cell 2 Line 5_
```markdown
# Requirements

For this example, you will need:

- An Elastic deployment with minimum **4GB machine learning node**
```
We use Alibaba's Cloud right? So we dont need a ML Node here as the work is proxied through to Alibaba

Comment (Huaixinww) [reply notebooks/alibabacloud-ai-search/inference-alibabacloud-ai-search.ipynb:1]: Thank you for your suggestion! I have removed the restriction on the 4GB machine learning node.

File: notebooks/alibabacloud-ai-search/inference-alibabacloud-ai-search.ipynb
Comment (Huaixinww) [reply notebooks/alibabacloud-ai-search/inference-alibabacloud-ai-search.ipynb:97]: Thanks for the suggestion!  I'v used `getpass` to store the hosts for improving security.

File: notebooks/alibabacloud-ai-search/inference-alibabacloud-ai-search.ipynb
Comment (joemcelroy) [inline notebooks/alibabacloud-ai-search/inference-alibabacloud-ai-search.ipynb:1]: _Commented on notebook `notebooks/alibabacloud-ai-search/inference-alibabacloud-ai-search.ipynb` Cell 2 Line 7_
```markdown
# Requirements

For this example, you will need:

- An Elastic deployment:
  - we'll be using [Alibaba Elasticsearch](https://www.aliyun.com/product/bigdata/elasticsearch) for this example.
- Elasticsearch 8.16 or above
```
is this correct as 8.16 has not been released yet. SHould this be 8.15 or is this a notebook that should be published on release of 8.16?

Comment (Huaixinww) [reply notebooks/alibabacloud-ai-search/inference-alibabacloud-ai-search.ipynb:1]: This notebook is related to the following pull request: https://github.com/elastic/elasticsearch/pull/111181, which is labeled as version 8.16.0. The inference API in version 8.15 does not support the Alibaba Cloud AI Search Model, so I believe this notebook should be published with the release of version 8.16.

Comment (joemcelroy) [reply notebooks/alibabacloud-ai-search/inference-alibabacloud-ai-search.ipynb:1]: OK great. Approving for now but will add a note that this is on hold to merge until 8.16 release cc @JessicaGarson .

**Review by joemcelroy — APPROVED** [review]

**Comment by Huaixinww** [comment]
> Ah you also need to run pre-commit, to format the notebook https://github.com/elastic/elasticsearch-labs/blob/main/CONTRIBUTING.md#pre-commit-hook

Thank you for the reminder! I've already run the pre-commit hook to format the notebook. Let me know if you need anything else!

**Comment by JessicaGarson** [comment]
This notebook isn't passing some of the checks but I'm happy to merge this after. I've found this to be helpful in the past when some of my code was in a similar state: 
 
```
python -m venv .venv
.venv/bin/pip install -qqq -r requirements-dev.txt
.venv/bin/pre-commit install
```

@Huaixinww feel free to ping me if you need any help here.

**Comment by Huaixinww** [comment]
> This notebook isn't passing some of the checks but I'm happy to merge this after. I've found this to be helpful in the past when some of my code was in a similar state:
> 
> ```
> python -m venv .venv
> .venv/bin/pip install -qqq -r requirements-dev.txt
> .venv/bin/pre-commit install
> ```
> 
> @Huaixinww feel free to ping me if you need any help here.

thanks a lot for ur suggestion, I've already run the pre-commit, the pre-commit check is successful. This notebook isn't passing some of the checks because I didn't set the `ELASTIC_USER` and `ELASTIC_PASSWORD` in my demo, I want to know how to pass this check? should I add a fake `ELASTIC_USER` and `ELASTIC_PASSWORD`?

**Comment by joemcelroy** [comment]
because this notebook relies on an alibaba account, we cannot rely on our automated tests to validate this notebook from regressions. This notebook needs to be excluded in https://github.com/elastic/elasticsearch-labs/blob/main/bin/find-notebooks-to-test.sh#L3

@JessicaGarson worth adding this in the contribute guide for when contributing notebooks to notebooks folder.

**Comment by joemcelroy** [comment]
pushed up to exclude it from the CI test script and also moved the notebook into integrations.

**Pull request merged**
